### PR TITLE
Enhancements in Do's and Don'ts game

### LINKED ISSUE #156: Enhancements to do's and don'ts game
## Description
As a [USER],
I need to modify do's and don'ts game,
so that I can play it with more difficulty levels too.

### Update [Required]
- [ ] Add separate lists of questions for different levels.
- [ ] Allow the user to press the submit button to submit.
- [ ] Creating a reset button so that its easy to reset all the available options. 

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
4-5 hours

### PR DESCRIPTION
**Pull request by lunayach**
### Description
The user is now able to change his answer any number of times before finally submitting. He is also able to reset his choices.

Fixes #156 

### Type of Change:

- Code
- User Interface
- Outreach

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
All Karma and Mocha tests passed.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 1]
```diff
@@ -35,24 +41,24 @@ export class DragdropComponent implements OnInit {
...
 
     /**
      *Empty Array of Objects which populates as the descriptions are dorpped into the Dos/ Donts Box
      */
-    do: { description: String, value: String }[]  = [];
-    dont: { description: String, value: String }[]  = [];
+    do: { description: string, value: string }[]  = [];
+    dont: { description: string, value: string }[]  = [];
 
     box1 = 'do';
```
Comment (shreyans29) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: convert these to constants, use them in the code below.

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 2]
```diff
@@ -62,6 +68,10 @@ export class DragdropComponent implements OnInit {
           this.language = response.pcprepkit.stages.malaria101.dragdrop;
           this.alerts = response.pcprepkit.common.alerts;
       });
+      this.dosAndDonts.forEach( (element) => {
+        (element.value === 'do') ? this.answerDo += element.description + ' ' : this.answerDont += element.description + ' ';
```
Comment (shreyans29) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: use the constant from above here.

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 3]
```diff
@@ -72,17 +82,31 @@ export class DragdropComponent implements OnInit {
...
         }
-            $box === 'do' ? this.do.push($event.dragData) : this.dont.push($event.dragData);
-            this.dosAndDonts.splice(this.dosAndDonts.indexOf($event.dragData), 1);
-            if (!this.dosAndDonts.length) {
-                this.onComplete();
-            }
-     }
+      }
 
+      if ($box === 'dont') {
```
Comment (shreyans29) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: same thing to be done throughout the file.

File: src/app/malaria-101/activity-2/activity-2.component.scss [hunk 4]
```diff
@@ -21,3 +22,18 @@
     border-width: 2px;
     border-style: dashed;
 }
+
+#submit-reset {
+    position: absolute;
+    left: $button-left-margin;
+}
+
+#submit {
```
Comment (Raxerz) [inline src/app/malaria-101/activity-2/activity-2.component.scss]: Now that you are using a class for the submit and reset buttons, are these css blocks required

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 5]
```diff
@@ -72,17 +82,31 @@ export class DragdropComponent implements OnInit {
      */
 
     addTobox($event: any, $box: String) {
-        this.reset();
-        if ($event.dragData.value === $box) {
-          this.score += this.POINTS_PER_CORRECT_ANSWER;
+      this.reset();
+      let pointsPerCorrectAnswer = this.POINTS_PER_CORRECT_ANSWER;
```
Comment (Raxerz) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: rename this (pointsPerCorrectAnswer) local variable to 'score' or 'points' or 'cumulativePoints'. It is confusing if it has the same name as the constant (POINTS_PER_CORRECT_ANSWER). (upper case or lower case does not help)

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 6]
```diff
@@ -108,14 +135,22 @@ export class DragdropComponent implements OnInit {
         }
       });
         this.activityComplete = true;
-        this._sharedData.customSuccessAlert(this.alerts.activitySuccessMsg, this.alerts.activitySuccessTitle);
+        swal({
```
Comment (Raxerz) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: custom swal alert from shared.data.service.ts can be used right?

**Comment by Raxerz** [comment]
@lunayach please fix the conflicts and address all the comments on this PR

**Comment by Raxerz** [comment]
@lunayach address the comments and fix conflicts as soon as possible

**Comment by lunayach** [comment]
Addressed comments.

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 7]
```diff
@@ -137,7 +165,12 @@ export class DragdropComponent implements OnInit {
         }
       });
         this.activityComplete = true;
-        this._sharedData.customSuccessAlert(this.alerts.activitySuccessMsg, this.alerts.activitySuccessTitle);
+        this._sharedData.customAlert(this.language.answerMessage, this.Solutions, 'info');
+        // swal({
```
Comment (paavininanda) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: Commented out code should not be there. Remove this if not needed.

File: src/app/malaria-101/activity-2/activity-2.component.scss [hunk 8]
```diff
@@ -1,5 +1,8 @@
 @import "../../styles/main";
 
+//Constants
+$button-left-margin: 46%;
+
 .available-item{
```
Comment (shreyans29) [inline src/app/malaria-101/activity-2/activity-2.component.scss]: spacing here..

File: src/app/malaria-101/activity-2/activity-2.component.ts [hunk 9]
```diff
@@ -17,10 +17,17 @@ export class DragdropComponent implements OnInit {
     public dropCheckDo: String;
     public dropCheckDont: String;
     language: any;
-    private score = 0;
+    public score = 0;
```
Comment (shreyans29) [inline src/app/malaria-101/activity-2/activity-2.component.ts]: it is a good practice to keep it private and write a public method say getScore, which will return the score value when ever you call.

**Comment by lunayach** [comment]
Addressed comments.

**Comment by Raxerz** [comment]
@lunayach if I add one correct answer to do and make the rest of it wrong, i get a score of 0.
Is that expected?

**Comment by lunayach** [comment]
@Raxerz there is a provision of negatives for wrong choices. If your score is < 0, you get 0.

**Pull request merged**